client/my-sites/checkout/composite-checkout/lib/submit-wpcom-transaction.ts

```diff
@@ -57,5 +59,14 @@ export default async function submitWpcomTransaction(
 		} );
 	}
 
-	return wp.req.post( '/me/transactions', mapRecordKeysRecursively( payload, camelToSnakeCase ) );
+	return wp.req.post(
+		'/me/transactions',
+		mapRecordKeysRecursively(
+			{
+				...payload,
+				tos: getToSAcceptancePayload(),
```

Comment:
This works fine, but I'm wary of adding properties to transaction data in this function because the data is not covered by types if it's changed once it gets here. Once we start modifying it, people are more likely to put additional changes to the data here.

Is there any possibility that we could add `tos` to the data just before this function is called? We even have functions to centralize that logic: `prepareRedirectTransaction()`, `prepareFreePurchaseTransaction()`, `createTransactionEndpointRequestPayload()`, etc.

Reply:
I think you make a fair point there and using `createTransactionEndpointRequestPayload()` looks like the best place to add the `tos` data. Thanks!

Reply:
👏

Reply:
Thanks! This inspired me to add comments to this function to make it more clear for future developers: https://github.com/Automattic/wp-calypso/pull/59888